charset/charset.go

```diff
@@ -57,15 +57,18 @@ var charsetInfos = map[string]*Charset{
 	CharsetASCII:   {CharsetASCII, CollationASCII, make(map[string]*Collation), "US ASCII", 1},
 	CharsetLatin1:  {CharsetLatin1, CollationLatin1, make(map[string]*Collation), "Latin1", 1},
 	CharsetBin:     {CharsetBin, CollationBin, make(map[string]*Collation), "binary", 1},
+	CharsetGBK:     {CharsetGBK, CollationGBKChineseCI, make(map[string]*Collation), "GBK", 2},
```

Comment:
This character set is supported by the switch in [PR](https://github.com/pingcap/tidb/pull/27288), so it should not be added directly here.
Related code: https://github.com/pingcap/tidb/pull/27288/files#diff-dd39cedcb0d828a897f39576501c11cceabb22408b0f0a08afe34574ea87fc4dR55